pkg/cmd/variable/shared/shared.go

```diff
@@ -0,0 +1,555 @@
+				return VariablePayload{}, err
+			}
+			values = append(values, data)
+
+			if data == Selected {
+				data, err = opts.Prompter.Input("Repos(comma separated)", getCurrentSelectedRepos(client.HTTP(), currentVar.SelectedReposURL))
+				if err != nil {
+					return VariablePayload{}, err
+				}
+				values = append(values, strings.Split(data, ",")...)
+			}
+		}
+
+		fmt.Fprintln(opts.IO.Out)
+		return getVarFromRow(opts, client, host, values, isUpdate)
+	}
+
+	body, err := io.ReadAll(opts.IO.In)
+	if err != nil {
+		return VariablePayload{}, fmt.Errorf("failed to read from standard input: %w", err)
+	}
+
+	return getVarFromRow(opts, client, host, strings.Split(opts.VariableName+","+string(bytes.TrimRight(body, "\r\n")), ","), isUpdate)
+}
+
+func GetVariablesForList(opts *ListOptions, getSelectedRepoInfo bool) ([]*Variable, error) {
+	client, err := opts.HttpClient()
+	if err != nil {
+		return nil, fmt.Errorf("could not create http client: %w", err)
+	}
+
+	orgName := opts.OrgName
+	envName := opts.EnvName
+
+	var baseRepo ghrepo.Interface
+	if orgName == "" {
+		baseRepo, err = opts.BaseRepo()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	variableEntity, err := GetVariableEntity(orgName, envName)
+	if err != nil {
+		return nil, err
+	}
+
+	variableApp, err := GetVariableApp(opts.Application, variableEntity)
+	if err != nil || variableApp == Unknown {
+		return nil, err
+	}
+
+	if !IsSupportedVariableEntity(variableApp, variableEntity) {
+		return nil, fmt.Errorf("%s variables are not supported for %s", variableEntity, variableApp)
+	}
+
+	var variables []*Variable
+
+	switch variableEntity {
+	case Repository:
+		variables, err = getRepoVariables(client, baseRepo, variableApp, opts.Page, opts.PerPage, opts.Name)
+	case Environment:
+		variables, err = getEnvVariables(client, baseRepo, envName, opts.Page, opts.PerPage, opts.Name)
+	case Organization:
+		var cfg config.Config
+		var host string
+
+		cfg, err = opts.Config()
+		if err != nil {
+			return nil, err
+		}
+
+		host, _ = cfg.DefaultHost()
+		variables, err = getOrgVariables(client, host, orgName, getSelectedRepoInfo, variableApp, opts.Page, opts.PerPage, opts.Name)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to get variables: %w", err)
+	}
+	return variables, nil
+}
+
+func getOrgVariables(client httpClient, host, orgName string, getSelectedRepoInfo bool, app App, page, perPage int, name string) ([]*Variable, error) {
+	variables, err := getVariables(client, host, fmt.Sprintf(`orgs/%s/%s/variables`, orgName, app), page, perPage, name)
+	if err != nil {
+		return nil, err
+	}
+
+	if getSelectedRepoInfo {
+		err = getSelectedRepositoryInformation(client, variables)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return variables, nil
+}
+
+func getEnvVariables(client httpClient, repo ghrepo.Interface, envName string, page, perPage int, name string) ([]*Variable, error) {
+	path := fmt.Sprintf(`repos/%s/environments/%s/variables`, ghrepo.FullName(repo), envName)
```

Comment:
Does this path work in APIs? The API docs have `/repositories/{repository_id}/environments/{environment_name}/variables` path

Reply:
yes, it does work.